extensions/azure/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndexExtension.java

```diff
@@ -54,6 +55,8 @@ public void initialize(ServiceExtensionContext context) {
         context.registerService(AssetLoader.class, assetIndex);
         context.registerService(DataAddressResolver.class, assetIndex);
 
+        context.getService(HealthCheckService.class).addReadinessProvider(() -> cosmosDbApi.get().forComponent(name()));
```

Comment:
I'm not that sure that a connection to an external data source should drive the readiness provider, it can lead to service fault and "connection refused" or "timeout" to the clients also for a temporary network failure, in my opinion the connector should handle those cases returning 500 and notifying the problem through the monitoring system.

Reply:
In a K8S environment, a failing `readiness` probe caused by transient errors will make the loadbalancer stop routing traffic to that pod, so this is indeed the intended behaviour, since the pod is not "ready to respond to requests" anymore.

However, if the application fails to start, e.g. due to a config error, that would be caught by the `startup` probe, which disables all other probes until it succeeds.

Monitoring and K8s probes have very different use cases.

Reply:
Indeed, and that is critical for resiliency (backpressure)